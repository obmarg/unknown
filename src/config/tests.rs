use std::{fs::File, io::Read};

use super::*;

#[test]
fn test_can_load_project_file() {
    let mut str_data = String::new();
    File::open("config-examples/project.kdl")
        .unwrap()
        .read_to_string(&mut str_data);

    insta::assert_debug_snapshot!(
        parse_project_file(&PathBuf::from("blah/project.kdl"), &str_data)
            .map_err(|e| miette::Report::new(e.0))
    )
}

#[test]
fn test_can_load_workspace_file() {
    let mut str_data = String::new();
    File::open("config-examples/workspace.kdl")
        .unwrap()
        .read_to_string(&mut str_data);

    insta::assert_debug_snapshot!(parse_workspace_file(
        &PathBuf::from("blah/workspace.kdl"),
        &str_data
    )
    .map_err(|e| miette::Report::new(e.0)))
}
