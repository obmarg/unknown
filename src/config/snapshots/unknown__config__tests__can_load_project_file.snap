---
source: src/config/tests.rs
expression: "parse_project_file(&source).map_err(|e| miette::Report::new(e.0))"
---
Ok(
    ProjectDefinition {
        project: "my_project",
        dependencies: DependencyBlock {
            projects: [
                ConfigPath {
                    span: SourceSpan {
                        offset: SourceOffset(
                            81,
                        ),
                        length: SourceOffset(
                            18,
                        ),
                    },
                    inner: "../other_project",
                },
            ],
        },
        tasks: TaskBlock {
            imports: [
                ConfigPath {
                    span: SourceSpan {
                        offset: SourceOffset(
                            521,
                        ),
                        length: SourceOffset(
                            11,
                        ),
                    },
                    inner: "/.tasks/*",
                },
                ConfigPath {
                    span: SourceSpan {
                        offset: SourceOffset(
                            574,
                        ),
                        length: SourceOffset(
                            16,
                        ),
                    },
                    inner: ".tasks/my_task",
                },
            ],
            tasks: [
                TaskDefinition {
                    name: "build",
                    commands: [
                        "cargo build",
                    ],
                    requires: [
                        TaskRequires {
                            task: "generate",
                            target: Some(
                                "^self",
                            ),
                        },
                    ],
                    input_blocks: [
                        InputBlock {
                            paths: [
                                Glob(
                                    Glob {
                                        glob: "whatever.txt",
                                        re: "(?-u)^whatever\\.txt$",
                                        opts: GlobOptions {
                                            case_insensitive: false,
                                            literal_separator: false,
                                            backslash_escape: true,
                                        },
                                        tokens: Tokens(
                                            [
                                                Literal(
                                                    'w',
                                                ),
                                                Literal(
                                                    'h',
                                                ),
                                                Literal(
                                                    'a',
                                                ),
                                                Literal(
                                                    't',
                                                ),
                                                Literal(
                                                    'e',
                                                ),
                                                Literal(
                                                    'v',
                                                ),
                                                Literal(
                                                    'e',
                                                ),
                                                Literal(
                                                    'r',
                                                ),
                                                Literal(
                                                    '.',
                                                ),
                                                Literal(
                                                    't',
                                                ),
                                                Literal(
                                                    'x',
                                                ),
                                                Literal(
                                                    't',
                                                ),
                                            ],
                                        ),
                                    },
                                ),
                                Glob(
                                    Glob {
                                        glob: "whatever/**",
                                        re: "(?-u)^whatever/.*$",
                                        opts: GlobOptions {
                                            case_insensitive: false,
                                            literal_separator: false,
                                            backslash_escape: true,
                                        },
                                        tokens: Tokens(
                                            [
                                                Literal(
                                                    'w',
                                                ),
                                                Literal(
                                                    'h',
                                                ),
                                                Literal(
                                                    'a',
                                                ),
                                                Literal(
                                                    't',
                                                ),
                                                Literal(
                                                    'e',
                                                ),
                                                Literal(
                                                    'v',
                                                ),
                                                Literal(
                                                    'e',
                                                ),
                                                Literal(
                                                    'r',
                                                ),
                                                RecursiveSuffix,
                                            ],
                                        ),
                                    },
                                ),
                            ],
                            env_vars: [
                                "xyz",
                            ],
                            commands: [
                                "",
                            ],
                        },
                    ],
                },
            ],
        },
    },
)
